app/actions/handlers.py

```diff
@@ -0,0 +1,166 @@
+
+    present_time = datetime.now(tz=timezone.utc)
+
+    try:
+        saved_state = await state_manager.get_state(str(integration.id), "pull_observations")
+        state = client.IntegrationState.parse_obj({"last_run": saved_state})
+    except pydantic.ValidationError as ve:
+        state = client.IntegrationState()
+
+    logger.info(f"Running Lotek integration for integration '{integration.name}({integration.id})'. State: {state}")
+
+    async for attempt in stamina.retry_context(on=(LotekException, LotekConnectionException), attempts=3, wait_initial=timedelta(seconds=10), wait_max=timedelta(seconds=10)):
+        with attempt:
+            auth = get_auth_config(integration)
+            try:
+                device_list = await client.get_devices(integration, auth)
+            except client.LotekException as e:
+                message = f"Error fetching devices from Lotek. Integration ID: {integration.id} Exception: {e}"
+                logger.exception(message)
+                await log_action_activity(
+                    integration_id=str(integration.id),
+                    action_id="pull_observations",
+                    title=message,
+                    level=LogLevel.WARNING
+                )
+                raise e
+
+    logger.info(f"Extracted {len(device_list)} devices from Lotek for inbound: {integration.id}")
+    observations_extracted = 0
+    for device in device_list:
+        cdip_positions = []
+        lower_date = max(present_time - timedelta(days=7), state.last_run)
+        while lower_date < present_time:
+            upper_date = min(present_time, lower_date + timedelta(days=7))
+            async for attempt in stamina.retry_context(on=(LotekException, LotekConnectionException), attempts=3, wait_initial=timedelta(seconds=10), wait_max=timedelta(seconds=10)):
+                with attempt:
+                    try:
+                        positions = await client.get_positions(device.nDeviceID, auth, integration, lower_date, upper_date, True)
+                    except httpx.HTTPError as e:
+                        message = f"Error fetching positions from Lotek. Device: {device.nDeviceID}. Dates: [{lower_date},{upper_date}]. Integration ID: {integration.id} Exception: {e}"
+                        logger.exception(message)
+                        await log_action_activity(
+                            integration_id=str(integration.id),
+                            action_id="pull_observations",
+                            title=message,
+                            level=LogLevel.WARNING
+                        )
+                        raise LotekException(message=message, error=e)
+            logger.info(f"Extracted {len(positions)} obs from Lotek for device: {device.nDeviceID} between {lower_date} and {upper_date}.")
+            for position in positions:
+                cdip_pos = transform(position, integration)
```

Comment:
@vgarcia13 this is where `cdip_pos` is not being assigned correctly. At the moment I think you're sending coroutine objects to the sensors API. If you were to change this to `await` that call, then it would probably be fine.

Like I mentioned in slack, please consider removing the `log_action_activity` call from your `transform` function, and make it sync. To avoid noise in the activity table. You can continue to log the message for troubleshooting, and maybe just keep track of how many are "bad lotek points" and publish one activity log message each run to record the count of these.